src/lib/sticky-header/sticky-header.spec.ts

```diff
@@ -0,0 +1,165 @@
+    stickyHeader = stickyElement.injector.get<CdkStickyHeader>(CdkStickyHeader);
+    scrollableElement = fixture.debugElement.query(By.directive(Scrollable));
+  });
+
+  it('should be able to find stickyParent', () => {
+    expect(stickyHeader.stickyParent).not.toBe(null);
+  });
+
+  it('should be able to find scrollableContainer', () => {
+    expect(stickyHeader.upperScrollableContainer).not.toBe(null);
+  });
+
+  it('should stick in the right place when scrolled to the top of the container', fakeAsync(() => {
+    let scrollableContainerTop = stickyHeader.upperScrollableContainer
+      .getBoundingClientRect().top;
+    expect(stickyHeader.element.getBoundingClientRect().top).not.toBe(scrollableContainerTop);
+    tick(0);
+
+    // Scroll the scrollableContainer up to stick
+    fixture.componentInstance.scrollDown();
+    tick(100);
+
+    expect(stickyHeader.element.getBoundingClientRect().top).toBe(scrollableContainerTop);
+  }));
+
+  it('should unstuck when scrolled off the top of the container', fakeAsync(() => {
+    let scrollableContainerTop = stickyHeader.upperScrollableContainer
+      .getBoundingClientRect().top;
+    expect(stickyHeader.element.getBoundingClientRect().top).not.toBe(scrollableContainerTop);
+    tick(0);
+
+    // Scroll the scrollableContainer up to stick
+    fixture.componentInstance.scrollDown();
+    tick(100);
+
+    expect(stickyHeader.element.getBoundingClientRect().top).toBe(scrollableContainerTop);
+
+    // Scroll the scrollableContainer down to unstuck
+    fixture.componentInstance.scrollBack();
+    tick(100);
+
+    expect(stickyHeader.element.getBoundingClientRect().top).not.toBe(scrollableContainerTop);
+
+  }));
+});
+
+describe('sticky-header with positioning supported', () => {
+  let fixture: ComponentFixture<StickyHeaderTest>;
+  let testComponent: StickyHeaderTest;
+  let stickyElement: DebugElement;
+  let stickyParentElement: DebugElement;
+  let stickyHeader: CdkStickyHeader;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [ OverlayModule, PlatformModule, StickyHeaderModule ],
+      declarations: [StickyHeaderTest],
+      providers: [
+        {provide: STICKY_HEADER_SUPPORT_STRATEGY, useValue: true},
+      ],
+    });
+    TestBed.compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(StickyHeaderTest);
+    fixture.detectChanges();
+    testComponent = fixture.debugElement.componentInstance;
+    stickyElement = fixture.debugElement.query(By.directive(CdkStickyHeader));
+    stickyParentElement = fixture.debugElement.query(By.directive(CdkStickyRegion));
+    stickyHeader = stickyElement.injector.get<CdkStickyHeader>(CdkStickyHeader);
+  });
+
+  it('should find sticky positioning is applied', () => {
+    let position = window.getComputedStyle(stickyHeader.element).position;
+    expect(position).not.toBe(null);
+    if (position != null) {
```

Comment:
Because in ` expect(/sticky/i.test(position)).toBe(true);`, the `test()` function's parameter can only be string. And the type of `position` is `string | null`, which will cause an error. S we need to add the `'!= null'` first.